lib/ClangImporter/SwiftLookupTable.cpp

```diff
@@ -1927,22 +1927,35 @@ void importer::addEntryToLookupTable(SwiftLookupTable &table,
   // possible to find these decls during deserialization. For any C++ typedef
   // that defines a name for a class template instantiation (e.g. std::string),
   // import the mangled name of this instantiation, and add it to the table.
+  auto addTemplateSpecialization =
+      [&](clang::ClassTemplateSpecializationDecl *specializationDecl) {
+        auto name = nameImporter.importName(specializationDecl, currentVersion);
+
+        // Avoid adding duplicate entries into the table.
+        auto existingEntries =
```

Comment:
This works because the name is mangled based on the specialization?

Reply:
Yeap, `nameImporter.importName` above will return the mangled name that starts with `__CxxTemplateInst...`.